csharp/src/SeedLang.Block/ArithmeticOperatorBlock.cs

```diff
@@ -21,7 +21,13 @@ public class ArithmeticOperatorBlock : OperatorBlock {
 
     public override void Accept(IBlockVisitor visitor) {
       visitor.VisitEnter(this);
+      visitor.VisitArithmeticOperatorBlock(this);
       visitor.VisitExit(this);
     }
+
+    protected override bool ValidateText(string text) {
+      // TODO: Validate the input text with the underlying parser.
```

Comment:
There are two use cases:

1. In a Unity UI app, the operator blocks are pre-defined and rendered. Users just select an operator block and drag it to an expression block to dock it there. Users can also click an existing operator block to open a drop-down menu, in order to change the operator. - in this case, we need a "valid operator list", instead of a "validator".

2. For the BxfParser, since the source code is in JSON text format, anyone can edit the source JSON. For example, if there is a **standalone** operator block and it has been successfully serialized to a JSON, then the JSON is mistakenly edited by someone (by changing "+" to "#" for example), the BxfReader need a validator to tell if the imported operator name is valid or not.

As for **verify** vs **validate**, I did a google search and found an IEEE definition in the following wiki page:

https://en.wikipedia.org/wiki/Verification_and_validation

* "Validation. The assurance that a product, service, or system meets the needs of the customer and other identified stakeholders. It often involves acceptance and suitability with external customers. Contrast with verification."
* "Verification. The evaluation of whether or not a product, service, or system complies with a regulation, requirement, specification, or imposed condition. It is often an internal process. Contrast with validation."

It looks that **verify** is the recommended word for our context.

I will change the wording in this pr.

Reply:
Wait... for the **verify** vs **validate** again, you said you used "verify" in the parser? I searched the current code base and found many "validate" cases in your Parser code (including PythonParser.Validate):

https://github.com/aha-001/SeedLang/search?q=validate

And there is no "verify" found in the codebase.

Are we simply keep the term pair "validate" and "validator"? I don't think IEEE's definition is a must-have rule.

Reply:
I was wrong with "verify". Maybe I planned to use "verify" in the beginning. And decided to use validate eventually.

Let's keep validate.